### DISCUSSION TITLE
Unable to set node name for JMX Mbeans

### DISCUSSION THREAD

**Discussion by Sulbigar** [xetorthio/jedis#3070] [Q&A]
Team,
I have a high traffic application that has more than 100 containers connecting to a redis cluster with 9 shards (1primary and 1 secondary). I'm trying to track the client pool connection metrics. 

When the GenericObjectPool gets initialized, the Mbean gets registered but one does not have control over setting the node name in the MBean key as the nodes are discovered in this process and I do not see a way to 
- intercept/callback during pool creation to update the jmx prefix with the discovered nodes or 
- deregister and re-register Mbeans by adding the discovered nodes.

What is the recommendation in this case? Without adding that identifier the pools are registered with name as jedisPool, jedisPool2, jedisPool3....and so on. [where jedisPool is the jmxPrefix]?

#### Jedis version:
3.7.1
#### Redis version:
6.2.5
#### Java version:
1.8

**Comment by sazzad16** [comment]
@Sulbigar Did you check if the feature is supported by [Commons Pool](https://github.com/apache/commons-pool) already? And if it's not supported yet, shouldn't commons pool be the proper place for this feature instead of Jedis?

**Reply by Sulbigar** [reply]
Thanks for the reply @sazzad16.
As jedis is abstracting the creation of GenericObjectPool, I feel it will be optimal to have a property like` "jmxNodeDiscovery"` which can then be utilized in `JedisClusterInfoCache` after discovering node to augment the `jmxPrefix` with a key for the JMX Object name like 
`"node=<nodehost>"` so that it will be used by the GenericObjectPool while registering for JMX.

**Comment by Sulbigar** [comment]
Any recommendations here? Would you prefer me raising a PR with this capability added?

**Reply by sazzad16** [reply]
@Sulbigar A PR usually helps to discuss further and visualize possible solution. But I would request you not to put much high hope after the PR.
